resources/aws/errors.go

```diff
@@ -64,24 +52,76 @@ func IsRouteNotFoundError(err error) bool {
 	return errgo.Cause(err) == routeNotFoundError
 }
 
-func IsSecurityGroupFind(err error) bool {
-	return errgo.Cause(err) == securityGroupFindError
+var securityGroupNotFoundError = errgo.New("security group not found")
+
+// IsSecurityGroupNotFound asserts securityGroupNotFoundError.
+func IsSecurityGroupNotFound(err error) bool {
```

Comment:
There are lots of not found errors. It would maybe make sense to have one "not found" error and just annotate error messages in the specific use case.